Lib/threading.py

```diff
@@ -800,8 +799,20 @@ class is implemented.
         assert group is None, "group argument must be None for now"
         if kwargs is None:
             kwargs = {}
+        if name:
+            name = str(name)
+        if not name:
+            name_template = "Thread-%d"
+            if target is not None:
+                try:
+                    target_name = target.__name__
+                    name_template = f"{name_template} ({target_name})"
+                except AttributeError:
+                    pass
+            name = _newname(name_template)
```

Comment:
It will fail if `target.__name__` contains `%` (please add a test for such case, and include also `{` just for the case).

It would be safer to write:
```python
name = _newname()
if target is not None:
    ...
    name = f"{name} ({target_name})"
```

Reply:
I wrote the code differently to avoid formatting errors.